docs/functional/architecture/0013-Scalar_Operators.md

```diff
@@ -0,0 +1,135 @@
+# Scalar operators
+
+How about "elementwise operators"?
+
+## What are scalar operators?
+
+Scalar operators are functions that take scalar arguments and return a scalar result. They can be used to implement arbitrary pointwise operations on fields.
+
+As an example, let's calculate the pointwise arithmetic-geometric mean of two fields:
+
+```python
+@scalar_operator
+def agm(x: float, y: float, tol: float) -> float:
+    an = x
+    gn = y
+    while abs(x - y) > tol:
+        an, gn = 0.5*(an + gn), math.sqrt(an * gn)
+    return an
+
+@field_operator
+def use_agm(x: Field[[D], float], y: Field[[D], float]) -> Field[[D], float]:
+    return agm(x, y)
+```
+
+In the code above, the function `agm` is executed once for every element pair of the `(x, y)` field pair, and a field of the same size as both `x` and `y` is produced. The function `agm` is essentially just a complicated pointwise binary operator, like `+` or `-`.
+
+
+## Why do we want scalar operators?
+
+Gt4py is concerned with the efficient parallelization of code, which is much easier to do when the source code can be turned into a simple data flow graph. To ensure this, the syntax of field operators is restricted and several Python constructs, such as loops, cannot be used.
+
+However, gt4py is not concerned with the optimziation of a single thread in the parallel grid, as that is delegated to platform compilers and has a smaller interference with parallel optimizations. Consequently, gt4py could allow fewer restrictions on the syntax in a scalar operator that describes only a single thread of execution. Unlike field operators, scalar operators can easily support conditionals, loops and mutable variables.
+
+Scalar operators can be used to implement features that, using the pure field operator syntax, would be difficult, impractical, or downright impossible. In weather and climate the general targets for scalar operators would be iterative algorithms and solvers. 
+
+## Implementation options
+
+### Within or outside the toolchain
```

Comment:
```suggestion
```
This is not needed. The lines below function as introduction to the next chapters